### PR TITLE
M600 R<resume temperature>

### PR DESCRIPTION
**Pull request by chestwood96**
### Description

I added the 'N' parameter to M600 to tell it the printing temperature of the new material.

With those changes M600 will unload like before, then heat either the current target temperature or the next target temperature (whichever is hotter) to load and purge. After the purge is done it will wait until it has reached the new target temperature and return printing.

If no N parameter is given it will behave like normal.

### Benefits

Different materials start printing at their set temperature instead of printing at the temperature of the previous one for a bit. May also be useful for the same material if the color or finish is temperature dependent.

The N parameter can be dynamically set using prusa slicer macros: 
`M600 N[temperature_[next_extruder]]; Filament change`

### CODE REVIEW

File: Marlin/src/feature/pause.cpp [hunk 1]
```diff
@@ -558,9 +558,17 @@ void resume_print(const float &slow_load_length/*=0*/, const float &fast_load_le
     thermalManager.reset_hotend_idle_timer(e);
   }
 
+  if (targetTemp > thermalManager.degTargetHotend(active_extruder))
+    thermalManager.setTargetHotend(targetTemp, active_extruder);
```
Comment (thinkyhead) [inline Marlin/src/feature/pause.cpp:566]: Simplified, since the temp is only set when above target.

Comment (chestwood96) [reply Marlin/src/feature/pause.cpp:566]: Good catch.

**Comment by AnHardt** [comment]
Really? 
'N' like in "Line**N**umber"?

**Comment by chestwood96** [comment]
Not quite sure how to read this comment.

Yes really, no not lineNumber, why line number, I am confused?

I did choose N (as in **N**extTemperature) because T (for **T**arget, **T**emperature or **T**arget**T**emperature) was allready taken. If I somehow violated a convention I have no issues about changing it.

**Comment by thinkyhead** [comment]
The `N` parameter is customarily verboten, along with `M` and `G`. There is only one other G-code that uses the `N` parameter, and that's kind of an oversight. A better parameter would be the first available out of `S`, `P`, `H`, `I`, etc. and if another "acronym" letter can be chosen that is preferred. So, instead of "(N)ext" temperature it could be "(R)esume" if that letter is available.

**Pull request merged**